### DISCUSSION TITLE
CORS enabling

### DISCUSSION THREAD

**Discussion by martijnlammers** [microsoft/CCF#3629] [Q&A] [answered]
Does CCF support enabling CORS server side and if so where would you need to enter valid domains?

**Comment by achamayou** [comment, marked as answer]
@martijnlammers my understanding of the implementation of CORS server side (derived from https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) is that it consists entirely of the server setting the right HTTP headers on responses. Since CCF lets you set arbitrary response headers, I would say it's possible to do it, eg:

```
ctx.rpc_ctx->set_response_header(
    http::headers::CONTENT_TYPE, http::headervalues::contenttype::JSON);
```

If you mean support as in, does CCF implement any CORS-related logic that can be switched on with a flag, then no.

**Reply by martijnlammers** [reply]
Alright, I'll give that a try, thank you.
